addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -160,4 +161,84 @@ options.registry.website_sale = options.Class.extend({
         }).then(this.reload);
     }
 });
+
+/**
+ * Handles the edition of snippet's products search bar.
+ */
+options.registry.productsSearchBar = options.Class.extend({
+    xmlDependencies: ['/website_sale/static/src/xml/website_sale.editor.xml'],
+    preventChildPropagation: true,
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.searchProductsBar = this.$target.find('.o_products_search_bar_input');
+        this.searchProductsInput = this.$target.find('.o_search_products');
```

Comment:
You have to call the _super method and return its value, check the other options.